### DISCUSSION TITLE
Accelsim memory footprint of completed kernels

### DISCUSSION THREAD

**Discussion by rvnonsec** [accel-sim/accel-sim-framework#350] [Ideas]
Hi,

I noticed that when i run the simulation with say kernel-2 alone  vs kernel-1 followed by kernel-2 in that order, the memory utilisation of the simulator when its running kernel-2 can increase significantly (like more than double in my case, was fitting easily in a 8GB VM vs barely running in a 16GB VM). 

IE, Some memory space related to the execution of the previous kernel is not freed even when kernel-2 is running. Obvioulsy, this can add up quickly if we run multiple kernels. Any ideas on what might be causing this bloat? Stats maybe? But stats feels too small to have sucha  large impact 
I can follow the suggestions/debug and try to help submit a PR for a change that fixes this and frees up the unnecessary memory. 

Thanks!

**Comment by JRPan** [comment]
Weird. This does not happen to us. I assume you used trace compression? Can you try without it? I wonder if this is caused by finished kernel traces not freed proporly.
